pkg/ebpf/state.go

```diff
@@ -160,6 +174,41 @@ func (ns *networkState) Connections(id string, latestTime uint64, latestConns []
 	return conns
 }
 
+func (ns *networkState) AddDNSStats(id string, conns []ConnectionStats) {
+	ns.Lock()
+	defer ns.Unlock()
+
+	if _, ok := ns.clients[id]; !ok {
+		return
+	}
+	seen := make(map[dnsKey]struct{})
+
+	for i := range conns {
+		conn := &conns[i]
+		if conn.DPort != 53 {
+			continue
+		}
+		key := dnsKey{
+			serverIP:   conn.Dest,
+			clientIP:   conn.Source,
+			clientPort: conn.SPort,
+			protocol:   conn.Type,
+		}
+
+		if _, alreadySeen := seen[key]; alreadySeen {
+			continue
+		}
+
+		if dnsStats, ok := ns.clients[id].dnsStats[key]; ok {
+			conn.DNSReplyCount = dnsStats.replies
+		}
+		seen[key] = struct{}{}
```

Comment:
can you delete from dnsStats instead of keeping a map for seen connections?

Reply:
Sure. The only reason for doing this way was to add code in the next iteration to detect collisions and probably update some metric (like we do for pid collisions when adding TCP stats). But if it doesn't sound like a metric that we would be interested in, I can directly delete from `dnsStats` as you suggested.